### PR TITLE
[CCI][#3600][BUG] Fixed timestamp with zeros after decimal point

### PR DESCRIPTION
**Pull request by frost017**
### Description
Before Dashboard was showing timestamp something like this: Mar 13, 2023 @ 18:13:13.**678000000**. Now Dashboard shows timestamp exactly as in the source(Mar 13, 2023 @ 18:13:13.678**668399**). For example the source timestamp: 
```
PUT my-index-000001
{
  "mappings": {
    "properties": {
      "timestamp": {
        "type": "date"
      }
    }
  }
}

POST my-index-000001/_doc/
{
  "abc": "def",
  "timestamp": "2023-03-13T13:13:13.678668399Z"
}

GET my-index-000001/_search
```

**Result:**

image

### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3600

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
  - [x] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by AMoo-Miki — CHANGES_REQUESTED** [review]
I wonder if we should take this behavior to `src/plugins/data/common/index_patterns` so every other function using it can benefit.

File: CHANGELOG.md [hunk 1]
```diff
@@ -109,8 +109,10 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [Region Maps] Add ui setting to configure custom vector map's size parameter([#3399](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3399))
 - [Search Telemetry] Fixes search telemetry's observable object that won't be GC-ed([#3390](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3390))
 - Clean up and rebuild `@osd/pm` ([#3570](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3570))
+- [Dashboard] Fixes timestamp with zeros after decimal point([#3684](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3684))
```
[the remark this answers is not in the document]
Comment (frost017) [reply CHANGELOG.md]: Hi! Thanks, changed it!

**Comment by frost017** [comment]
> I wonder if we should take this behavior to `src/plugins/data/common/index_patterns` so every other function using it can benefit.

Hi! 
Initially I also though about this, but as for me it seems to be a breaking change. I guess there are a lot places where methods from this file `src/plugins/data/common/index_patterns` are used. Also it may need thorough research to accomplish this safely.

**Comment by joshuarrrr** [comment]
@frost017 I'd start by tracking down all the places that use the `indexPattern.formatHit(hit, 'html')` method. Anywhere that's used, the timestamp formatting may be wrong. As far as I remember, the `html` formatter is not used as commonly, and may have some other formatting bugs.

**Comment by joshuarrrr** [comment]
@frost017 I'm marking this as a draft for now, until you have a chance to work on it some more.

**Review by kavilla — CHANGES_REQUESTED** [review]
Thank you @frost017 for this and I appreciate the work here. But I think the fix should be an indicator for the user. More information: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3600#issuecomment-1544401557

**Comment by frost017** [comment]
Hi! Thank you for suggestions! I am actively working on it

**Comment by kavilla** [comment]
Closing as `valued effort` this contribution will not be merged but still be recognized as a "contribution". If you would like to open another PR that provides in app assistance with this please feel free to do so @frost017 !

Sorry about this and thanks again!

**Pull request closed without merging**